tests/test_utils.py

```diff
@@ -163,62 +151,17 @@ def test_dot_product():
     assert dot_product([1, 2, 3], [0, 0, 0]) == 0
 
 
-def test_element_wise_product():
```

Comment:
Where did these functions go?

Reply:
This is an inefficient way to implement operations between matrices, I think that the focus of the book is the intelligent algorithms so operations between matrices of this type can be delegated to numpy that performs them in a more efficient manner.